src/requirements/data/templates/ece.ts

```diff
@@ -0,0 +1,86 @@
+import { Template } from '@/requirements/types';
+
+const eceTemplate: Template = [
+  {
+    reqGroup: 'UNI',
+    name: 'Physical Education',
+    placeholderSemesters: [1, 2],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Mathematics',
+    placeholderSemesters: [1, 2, 3, 4],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Physics',
+    placeholderSemesters: [2, 3, 4],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Chemistry',
+    placeholderSemesters: [1],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'First-Year Writing Seminars',
+    placeholderSemesters: [1, 2],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Advanced Programming/Computer Engineering',
+    placeholderSemesters: [3],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'ENGRD 2300',
+    placeholderSemesters: [3],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Core Courses',
+    placeholderSemesters: [3, 4],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Engineering Distribution',
+    placeholderSemesters: [4],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Liberal Studies: 6 courses',
+    placeholderSemesters: [3, 4, 5, 6, 7, 8],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Outside ECE Technical Electives',
+    placeholderSemesters: [5, 6, 7],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Culminating Design Experience (CDE)',
+    placeholderSemesters: [7],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Advisor Approved Electives',
+    placeholderSemesters: [3, 4],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Foundation Courses',
+    placeholderSemesters: [5, 6],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Upper-Level ECE Electives',
+    placeholderSemesters: [6],
+  },
+  {
+    reqGroup: 'EN',
+    name: 'Upper-Level ECE Electives: 4000 level',
+    placeholderSemesters: [7, 8],
```

Comment:
I think both the template and reqs data are wrong for ECE electives. I think there should be 3 upper-level ece electives (semesters 6, 7, 8), 2 upper-level ECE electives (both in semester 8), and 1 CDE course (accurate). If @einc agrees, could you update the major and the template?

![image](https://user-images.githubusercontent.com/25535093/142492780-f5b50433-8698-4ea8-8e14-3ba6dba2d6dd.png)

Reply:
Yes, the data in the major file seems to suggest 1 elective and 3 4000+ electives, which I tried to match. It does seem to be out of date however so I will update the major file too.

Reply:
After talking to @einc - it looks like for pre-2021 students we are going to want a total of 5 electives, but 6 electives for post 2021 (per the handbooks). We can talk about how to do this (toggleable requirements probably) after break!!